parquet/src/main/java/org/apache/iceberg/parquet/ParquetReader.java

```diff
@@ -135,11 +135,9 @@ private void advance() {
         throw new RuntimeIOException(e);
       }
 
-      long rowPosition = rowGroupsStartRowPos[nextRowGroup];
+      model.setPageSource(pages, rowGroupsStartRowPos == null ? 0 : rowGroupsStartRowPos[nextRowGroup]);
```

Comment:
The `rowPosition` will be ignored if the position column is not projected.

Reply:
I don't think this file should change. If the determination not to use correct start positions in made in `ReadConf`, then it should just set the position to 0 there. This can use whatever `rowGroupStartRowPos` contains.